streams/src/main/java/org/apache/kafka/streams/state/internals/InMemoryKeyValueLoggedStore.java

```diff
@@ -79,6 +79,15 @@ public void put(K key, V value) {
     }
 
     @Override
+    public void putIfAbsent(K key, V value) {
+        if (get(key) == null) {
+            this.inner.putIfAbsent(key, value);
+            changeLogger.add(key);
+            changeLogger.maybeLogChange(this.getter);
+        }
```

Comment:
I noticed now that `putIfAbsent` doesn't return anything. The typical signature is usually like:

https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ConcurrentMap.html#putIfAbsent-K-V-